### ISSUE TITLE
False positive: unpacking-non-sequence after return

### ISSUE THREAD

**Issue by eli-schwartz** [pylint-dev/pylint#9095]
### Bug description

```python
class DubDependency(ExternalDependency):
    # dub program and version
    class_dubbin: T.Optional[T.Tuple[ExternalProgram, str]] = None
    class_dubbin_searched = False

    def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.Any]):

[...]

        if DubDependency.class_dubbin is None:
            if self.required:
                raise DependencyException('DUB not found.')
            self.is_found = False
            return

        (self.dubbin, dubver) = DubDependency.class_dubbin ### <<< pylint error
```

After upgrading to pylint 3 in CI, this triggers the following error:

```
mesonbuild/dependencies/dub.py:56:8: E0633: Attempting to unpack a non-sequence defined at line 32 (unpacking-non-sequence)
```

Since there is a return when it is a non-sequence, it is safe to unpack.

### Command used

```shell
https://github.com/mesonbuild/meson/


pylint mesonbuild/

```

### Pylint output

```python
************* Module mesonbuild.dependencies.dub
mesonbuild/dependencies/dub.py:56:8: E0633: Attempting to unpack a non-sequence defined at line 32 (unpacking-non-sequence)
```

### Expected behavior

A clean bill of health.

### Pylint version

```shell
pylint 3.0.0
astroid 3.0.0
Python 3.11.5 (main, Sep 20 2023, 20:02:43) [GCC 12.3.1 20230526]
```
